geomesa-accumulo/geomesa-accumulo-datastore/src/main/scala/org/locationtech/geomesa/accumulo/index/QueryFilterSplitter.scala

```diff
@@ -223,7 +224,45 @@ class QueryFilterSplitter(sft: SimpleFeatureType) extends Logging {
       }
     }
 
-    makeDisjoint(mergeOverlappedFilters(combineSecondaryFilters(reduceChildOptions(getChildOptions))))
+    // detect the pattern it the pattern (attr in (1,2,3,4,5,6, etc) AND something else)
+    def detectSingleAttrOr: Boolean = {
+      // each child is expressed as (attr = x(i) AND expr2 AND expr3 ...)
+      val attrs =
+        filter.getChildren.map {
+          case and: And => and.getChildren.flatMap {
```

Comment:
also note that i think we may want to redesign some of the code anyways to make it easier to follow/inject new optimizations :)